src/main/java/org/spongepowered/api/entity/ai/Goal.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.ai;
+
+import org.spongepowered.api.entity.ai.task.AITask;
+import org.spongepowered.api.entity.ai.task.AITaskType;
+import org.spongepowered.api.entity.living.Agent;
+import org.spongepowered.api.entity.living.monster.Monster;
+
+import java.util.List;
+
+/**
+ * Represents a series of tasks meant to achieve a "goal".
+ *
+ * In Minecraft, most {@link Agent}s have just one goal which is living or simulating, or
+ * whichever term you want to call it. Other agents, such as {@link Monster}s, have a
+ * "target" goal which serves to seek out potential victims and execute a series of attack
+ * tasks.
+ *
+ * @param <O> The type of agent
+ */
+public interface Goal<O extends Agent> {
+
+    /**
+     * The type of this goal.
+     *
+     * @return The type
+     */
+    GoalType getType();
+
+    /**
+     * The {@link Agent} that owns this goal.
+     *
+     * @return The owner
+     */
+    O getOwner();
```

Comment:
Goals are the direct instances sitting on an Agent currently. It isn't the intention, at this time, to let you create custom goals (as that involves actually rigging those into the update loop).

I'll think about it.